common/src/commonMain/kotlin/Color.kt

```diff
@@ -1,62 +1,55 @@
-public class Color(rgb: Int) {
+public value class Color(private val packedRGB: Int) {
     public constructor(red: Int, green: Int, blue: Int) : this(rgb(red, green, blue))
 
-    public val rgb: Int = rgb and 0xFFFFFF
-
+    // removes alpha channel if present
+    public val rgb: Int get() = packedRGB and 0xFFFFFF
     public val red: Int get() = (rgb shr 16) and 0xFF
     public val green: Int get() = (rgb shr 8) and 0xFF
     public val blue: Int get() = (rgb shr 0) and 0xFF
 
     init {
-        require(this.rgb in MIN_COLOR..MAX_COLOR) { "RGB should be in range of $MIN_COLOR..$MAX_COLOR but was ${this.rgb}" }
+        requireInRange("RGB", rgb, acceptableColorRange)
     }
 
-    override fun toString(): String = "Color(red=$red,green=$green,blue=$blue)"
-
-    override fun hashCode(): Int = rgb.hashCode()
-
-    override fun equals(other: Any?): Boolean {
-        val color = other as? Color ?: return false
-
-        return color.rgb == rgb
-    }
+    override fun toString(): String = "Color(red=$red, green=$green, blue=$blue)"
 
     public companion object {
         private const val MIN_COLOR = 0
         private const val MAX_COLOR = 0xFFFFFF
+        private val acceptableComponentRange = 0..255
+        private val acceptableColorRange = MIN_COLOR..MAX_COLOR
 
         private fun rgb(red: Int, green: Int, blue: Int): Int {
-            require(red in 0..255) { "Red should be in range of 0..255 but was $red" }
-            require(green in 0..255) { "Green should be in range of 0..255 but was $green" }
-            require(blue in 0..255) { "Blue should be in range of 0..255 but was $blue" }
-
+            requireInRange("Red", red, acceptableComponentRange)
+            requireInRange("Green", green, acceptableComponentRange)
+            requireInRange("Blue", blue, acceptableComponentRange)
 
             return red and 0xFF shl 16 or
-                (green and 0xFF shl 8) or
-                (blue and 0xFF) shl 0
+                    (green and 0xFF shl 8) or
+                    (blue and 0xFF) shl 0
+        }
+
+        private fun requireInRange(name: String, value: Int, range: IntRange) {
+            require(value in range) { "$name should be in range of $range but was $value" }
         }
     }
 
     internal object Serializer : KSerializer<Color> {
-        override val descriptor: SerialDescriptor
-            get() = PrimitiveSerialDescriptor("Kord.color", PrimitiveKind.INT)
-
-        override fun deserialize(decoder: Decoder): Color = Color(decoder.decodeInt())
-
-        override fun serialize(encoder: Encoder, value: Color) {
-            encoder.encodeInt(value.rgb)
-        }
+        private val delegate = Int.serializer()
+        override val descriptor: SerialDescriptor = delegate.descriptor
+        override fun deserialize(decoder: Decoder): Color = Color(delegate.deserialize(decoder))
+        override fun serialize(encoder: Encoder, value: Color) = delegate.serialize(encoder, value.rgb)
```

Comment:
I think we no longer need a custom serializer now, right?

Reply:
Unfortunately, we need it here. We need to (de)serialize color without alpha, but backed value contains it (`packedRGB` prop). Alpha removes only by triggering `rgb` prop getter.